specification/_types/common.ts

```diff
@@ -339,3 +339,9 @@ export enum SlicesCalculation {
    */
   auto
 }
+
+export type ClusterInfoTarget = 'http' | 'ingest' | 'thread_pool' | 'script'
```

Comment:
This seems like a good candidate for an [enum](https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#enum).
It could encapsulate all the possibilities and `ClusterInfoTargets` would only be the union of the enum and an array of the same type.

Reply:
Hi @Anaethelion! In [this commit](https://github.com/elastic/elasticsearch-specification/pull/2140/commits/72fa7e6ca8e37bc66c335b4025e1a6da2b910af0) I changed to a `type` because the value could be `_all OR <one value of ClusterInfoTarget> OR <two or more values of ClusterTargetInfo>`. Defining it as an `enum` made it fail with the following [error](https://github.com/elastic/elasticsearch-specification/actions/runs/5199181765/jobs/9376248950)

Reply:
Look at this:

![Screenshot 2023-06-08 at 10 31 21](https://github.com/elastic/elasticsearch-specification/assets/284537/ac3c532d-0d6a-41ce-96b1-d1d4dc9fec5c)

it does work, but the code looks _odd_ to me. You can use a `string` or a value coming from an `enum`  mmh... this is clearly personal, but it does not look _pretty_ to me.

On the other hand, using the type union of strings, the API would look like:
![Screenshot 2023-06-08 at 10 34 07](https://github.com/elastic/elasticsearch-specification/assets/284537/8fed1779-5e84-40a1-92d2-3af05203ff9d)

what is prettier –again, according to me.

I just want to show the potential options, I've no idea which one is better here.

Reply:
nevermind all that I've said. The error was because I used double-quotes 🤦🏽 
pushed the changes. Thanks!

Reply:
understood. changed it